pkg/param/render.go

```diff
@@ -119,11 +120,26 @@ func renderStringArg(arg string, tp TemplateParams) (string, error) {
 	}
 	buf := bytes.NewBuffer(nil)
 	if err = t.Execute(buf, tp); err != nil {
+		// Check if Error is because of undefined key,
+		// which will lead on execute error on first undefined (missingkey=error).
+		// Get the undefined key name from the error message.
+		if strings.Contains(err.Error(), "map has no entry for key ") {
+			errMsg := NewErrorMessage(err.Error(), "map has no entry for key ")
+			return "", errors.WithStack(errors.New(errMsg))
+		}
 		return "", errors.WithStack(err)
 	}
 	return buf.String(), nil
 }
 
+func NewErrorMessage(err string, value string) string {
```

Comment:
The name is not appropriate. This is not a generic function but can be used only with the error above. I would recommend using something more specific. `newUndefinedKeyError(err)`. No need to export the function and no need to pass in the `map has no entry for key` since we are changing it to a string const above.
Also, it can directly return `errors.WithStack` instead of a message.